File: BackEnd/src/domain/interfaces/usecases/chat/IGetMessagesUseCase.ts
/**
 * @file IGetMessagesUseCase.ts
 * @description Interfaz del caso de uso para obtener mensajes de una conversación.
 * @module Chat
 * @layer Domain
 */

import type { MessageResponseDTO } from '../../../dtos/chat/message-response.dto.js';

/**
 * Contrato del use case GetConversationMessages.
 * Verifica que el usuario es participante antes de devolver los mensajes.
 */
export interface IGetMessagesUseCase {
  /**
   * Ejecuta el caso de uso.
   *
   * @param conversationId — ID de la conversación.
   * @param userId — ID del usuario solicitante (para verificar participación).
   * @param page — Número de página (base 1). Por defecto 1.
   * @param limit — Elementos por página. Por defecto 30.
   * @returns Lista paginada de MessageResponseDTOs ordenada por fecha descendente.
   * @throws {NotFoundException} Si la conversación no existe.
   * @throws {ForbiddenException} Si el usuario no es participante.
   */
  execute(conversationId: string, userId: string, page?: number, limit?: number): Promise<MessageResponseDTO[]>;
}

File: FrontEnd/lib/domain/interfaces/usecases/chat/i_get_messages_use_case.dart
/// @file i_get_messages_use_case.dart
/// @description Interfaz: Obtiene los mensajes paginados de una conversación.
/// @module Chat
/// @layer Domain
library;

import '../../../entities/message.dart';

// ═══════════════════════════════════════════════════════════════════════════════
// I GET MESSAGES USE CASE
// ═══════════════════════════════════════════════════════════════════════════════

/// Contrato del use case que recupera mensajes de una conversación con paginación.
abstract interface class IGetMessagesUseCase {
  /// Devuelve los mensajes de [conversationId], paginados.
  ///
  /// [param] conversationId — ID de la conversación.
  /// [param] page  — número de página (1-based). Por defecto 1.
  /// [param] limit — mensajes por página. Por defecto 30.
  /// [returns] lista de [Message] en orden cronológico ascendente.
  /// [throws] AppError — si la carga falla.
  Future<List<Message>> execute(
    String conversationId, {
    int page  = 1,
    int limit = 30,
  });
}
